extensions/amp-analytics/0.1/requests.js

```diff
@@ -36,13 +36,16 @@ const BATCH_INTERVAL_MIN = 200;
 
 export class RequestHandler {
   /**
-   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!Element} ampAnalyticsElement
    * @param {!JsonObject} request
    * @param {!../../../src/preconnect.Preconnect} preconnect
    * @param {function(string, !JsonObject)} handler
    * @param {boolean} isSandbox
    */
-  constructor(ampdoc, request, preconnect, handler, isSandbox) {
+  constructor(ampAnalyticsElement, request, preconnect, handler, isSandbox) {
+
+    /** @type {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc */
+    const ampdoc = ampAnalyticsElement.getAmpDoc();
 
     /** @const {!Window} */
     this.win = ampdoc.win;
```

Comment:
nit: can you change it to `this.win = ampAnalyticsElement.getAmpDoc().win`, and remove the `const ampdoc` instead.

Reply:
Done